Documentation/design-docs/unloadability.md

```diff
@@ -0,0 +1,111 @@
+﻿# `AssemblyLoadContext` unloadability
+## Goals
+* Provide a building block for the AppModel team to implement unloadable plug-ins
+* Users can load an assembly and its dependencies into an unloadable `AssemblyLoadContext`. 
+* There can be multiple unloadable `AssemblyLoadContext` instances coexisting
+* Users can unload the `AssemblyLoadContext` when there are no outside hard references to the types and instance of types from the assemblies loaded into the context and the assemblies themselves.
+* The unload can be initiated eagerly using explicit `Unload` method or it can happen after all references to the `AssemblyLoadContext` and all the assemblies in it are gone.
+## Non-goals
+* Forceful unload using thread abort or similar measures to cause threads to stop running with frames of code from the context to be unloaded on the call stack.
+## General scenarios
+Based on various discussions and feedback on github, the following general scenarios were often mentioned as use cases for unloadability.
+* Loading assemblies for one-off introspection. 
```

Comment:
I would put these 3 in opposite order. (Or at least make the introspection last since it is the least interesting scenario for collectible assemblies).